stripe/multipart_data_generator.py

```diff
@@ -21,7 +21,9 @@ def add_params(self, params):
             self._write(self.param_header())
             self._write(self.line_break)
             if hasattr(value, 'read'):
-                filename = value.name if hasattr(value, 'name') else "blob"
+                filename = "blob"
+                if hasattr(value, 'name'):
+                    filename = str(value.name)
```

Comment:
This fixes an edge case when using a `tempfile.TemporaryFile` instance: the `name` attribute exists but is an `int`, which caused an exception when trying to insert the filename into the buffer.

Reply:
Mind adding a short comment that says what you said here and what conditions we expect this to occur under?